packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -388,6 +407,10 @@ abstract class ProjectBase extends Resource implements IProject {
       ...props,
     });
   }
+
+  public bindToCodePipeline(_scope: Construct, _options: BindToCodePipelineOptions): void {
```

Comment:
Pass in the pipeline itself and expose the artifacts bucket as a readonly property from the pipeline if it’s not already there.

Reply:
That won't work. This might be either the artifact's main bucket, or a replication bucket if this project is used in a cross-region fashion. It's the same reason `bucket` is a property on `ActionBindOptions` in `IAction`, instead of being accessed through `stage.pipeline.artifactBucket`.